src/Serilog.Extensions.Logging/Extensions/Logging/EventIdPropertyCache.cs

```diff
@@ -0,0 +1,86 @@
+// Copyright (c) Serilog Contributors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+namespace Serilog.Extensions.Logging
+{
+    using System.Collections.Concurrent;
+    using Microsoft.Extensions.Logging;
+    using Serilog.Events;
+
+    static class EventIdPropertyCache
+    {
+        const int MaxCachedProperties = 1024;
+
+        static readonly ConcurrentDictionary<EventKey, LogEventProperty> s_propertyCache = new();
+        static int s_count;
+
+        public static LogEventProperty GetOrCreateProperty(in EventId eventId)
```

Comment:
With the class converted to non-static, it'd be possible to test that this method returns the same instance (`ReferenceEquals()`) for the same `eventId` properties, and different instances when different event ids are passed/when the cache is full.